### PR TITLE
[STORM-2363]Provide configuration to set the number of RollingWindow

### PR DESCRIPTION
**Pull request by liu-zhaokun**
> https://issues.apache.org/jira/browse/STORM-2363
I make the number of RollingWindow variable by add configuration which named num.stat.buckets.So users need't modify the number of RollingWindow with compiling storm-core.

### CODE REVIEW

**Comment by liu-zhaokun** [comment]
Hi,@HeartSaVioR 
Have you reviewed my PR?Looking forward to your reply.Thanks.

**Comment by HeartSaVioR** [comment]
@liu-zhaokun 
Sorry we're on release vote for 1.1.0 and personally I would like to postpone everything not related to release. Please stay tuned from dev@ mailing list.

**Comment by liu-zhaokun** [comment]
@HeartSaVioR 
OK!Thanks for your reply.

File: storm-core/src/jvm/org/apache/storm/Config.java [hunk 1]
```diff
@@ -2342,6 +2342,9 @@
     @isPositiveNumber
     public static String STORM_CGROUP_MEMORY_LIMIT_TOLERANCE_MARGIN_MB = "storm.cgroup.memory.limit.tolerance.margin.mb";
 
+	@isInteger
```
Comment (HeartSaVioR) [inline storm-core/src/jvm/org/apache/storm/Config.java]: Indentation is off.

File: storm-core/src/jvm/org/apache/storm/stats/CommonStats.java [hunk 2]
```diff
@@ -23,10 +23,11 @@
 import org.apache.storm.metric.api.IMetric;
 import org.apache.storm.metric.internal.MultiCountStatAndMetric;
 import org.apache.storm.metric.internal.MultiLatencyStatAndMetric;
+import org.apache.storm.utils.Utils;
 
 @SuppressWarnings("unchecked")
 public abstract class CommonStats {
-    public static final int NUM_STAT_BUCKETS = 20;
+    public static final int NUM_STAT_BUCKETS =(int) Utils.readStormConfig().get("num.stat.buckets");
```
Comment (HeartSaVioR) [inline storm-core/src/jvm/org/apache/storm/stats/CommonStats.java]: Could we find other approach, like passing that value from caller of CommonStats?
I think we're reading storm config file again and again which looks redundant, and even doing it for constant doesn't look good.

**Comment by liu-zhaokun** [comment]
@HeartSaVioR 
OK!I will try to find a more appropriate way.Thanks for your advice.

**Comment by HeartSaVioR** [comment]
@liu-zhaokun Let's close this PR.

**Pull request closed without merging**